src/engine/source/builder/builders/opBuilderHelperNetInfoAddress.cpp

```diff
@@ -0,0 +1,230 @@
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+base::Expression opBuilderHelperNetInfoAddress(const std::any& definition, bool isIPv6)
+{
+    auto [targetField, name, rawParameters] = helper::base::extractDefinition(definition);
+    auto parameters = helper::base::processParameters(rawParameters);
+
+    // Assert expected number of parameters
+    helper::base::checkParametersSize(parameters, 4);
+    // Parameter type check
+    // Agent_id
+    helper::base::checkParameterType(parameters[0],
+                                     helper::base::Parameter::Type::REFERENCE);
+    // scan_id
+    helper::base::checkParameterType(parameters[1],
+                                     helper::base::Parameter::Type::REFERENCE);
+    // name
+    helper::base::checkParameterType(parameters[2],
+                                     helper::base::Parameter::Type::REFERENCE);
+    // array (IPv4 or IPv6)
+    helper::base::checkParameterType(parameters[3],
+                                     helper::base::Parameter::Type::REFERENCE);
+
+    name = helper::base::formatHelperFilterName(name, targetField, parameters);
+
+    // Tracing
+    const auto successTrace = fmt::format("[{}] -> Success", name);
+    const auto failureTrace = fmt::format(
+        "[{}] -> Failure: Parameter doesn't exist or it has the wrong type: ", name);
+    const auto failureTrace1 = fmt::format(
+        "[{}] -> Failure: [{}] sysNetAddressTableFill error", name, targetField);
+    const auto failureTrace2 = fmt::format(
+        "[{}] -> Failure: [{}] couldn't assign result value", name, targetField);
+
+    // EventPaths and mappedPaths can be set in buildtime
+    auto wdb = std::make_shared<wazuhdb::WazuhDB>(STREAM_SOCK_PATH);
+
+    // Return Term
+    return base::Term<base::EngineOp>::create(
+        name,
+        [=,
+         targetField = std::move(targetField),
+         agent_id_path = parameters[0].m_value,
+         scan_id_path = parameters[1].m_value,
+         name_path = parameters[2].m_value,
+         ipObject_path = parameters[3].m_value,
+         wdb = std::move(wdb)](base::Event event) -> base::result::Result<base::Event>
+        {
+            // Checking values and saving them
+            if (!event->exists(agent_id_path) || !event->isString(agent_id_path))
+            {
+                return base::result::makeFailure(event, failureTrace + agent_id_path);
+            }
+            const auto& agent_id = event->getString(agent_id_path).value_or("NULL");
+
+            if (!event->exists(scan_id_path) || !event->isString(scan_id_path))
+            {
+                return base::result::makeFailure(event, failureTrace + scan_id_path);
+            }
+            const auto& scan_id = event->getString(scan_id_path).value_or("NULL");
+
+            if (!event->exists(name_path) || !event->isString(name_path))
+            {
+                return base::result::makeFailure(event, failureTrace + name);
```

Comment:
Was this condition tested? `name` might not be accessible as to how it is captured.

Reply:
I've replaced name by name_path that was the correct issue, and adding a test regarding this.